### PR TITLE
fix(Picking): fix issue with picking and widgets

### PR DESCRIPTION
**Pull request by martinken**
When using the hardware selector and widgets that are
in a different renderer (such as the orientation widget)
there was an issue with picking where the widget's renderer
was not being used.

### CODE REVIEW

**Review by floryst — APPROVED** [review]
LGTM.

Would it be helpful to deprecate `renderWindow.getSelector()` in favor of `createSelector()` to avoid potential sharing issues (like [this post](https://discourse.vtk.org/t/interactiveorientationwidget-loses-interactivity-once-hardwareselector-is-used/7702)) in the future?

**Comment by martinken** [comment]
> 

Let me think about this a bit more. Really we only need one selector and the renderer gets passed to it when capturing buffers. I think I just never updated the WidgetManager to use the new async API. Let me give that a shot.

**Comment by martinken** [comment]
OK I think this is an approach that will work with the async hardware selection. I'm not 100% sure of all the gotchas as async is always a bit tricky, but local testing on both backends seems to work.

**Comment by jourdain** [comment]
Might be worth checking the API call made to vtk.js for picking in [react-vtk-js](https://github.com/Kitware/react-vtk-js/blob/master/src/core/View.js) and [vue-vtk-js](https://github.com/Kitware/vue-vtk-js/blob/master/src/core/VtkView/script.js)

**Comment by martinken** [comment]
Do you know if either of those systems use the WidgetManager? The two links just use the selector which should be fine as the deprecation is in the WidgetManager.

**Comment by jourdain** [comment]
No widget, just plain hardware selector. Great to know... Sorry, I'm only half following. ;-)

File: Sources/Widgets/Core/WidgetManager/index.js [hunk 1]
```diff
@@ -293,11 +301,10 @@ function vtkWidgetManager(publicAPI, model) {
       subscriptions.pop().unsubscribe();
     }
 
-    model.selector = model.apiSpecificRenderWindow.getSelector();
+    model.selector = model.apiSpecificRenderWindow.createSelector();
```
Comment (floryst) [inline Sources/Widgets/Core/WidgetManager/index.js:304]: Do we still need to create a new selector, or can we use the existing one?

File: Sources/Widgets/Core/WidgetManager/index.js [hunk 2]
```diff
@@ -440,7 +451,50 @@ function vtkWidgetManager(publicAPI, model) {
...
+        await captureBuffers(x, y, x, y);
+      }
+
+      model.selections = model._capturedBuffers.generateSelection(x, y, x, y);
+    }
+    return publicAPI.getSelectedData();
+  };
+
   publicAPI.updateSelectionFromXY = (x, y) => {
+    console.warn(
```
Comment (floryst) [inline Sources/Widgets/Core/WidgetManager/index.js:495]: Maybe use `vtkWarningMacro` here.

File: Sources/Widgets/Core/WidgetManager/index.js [hunk 3]
```diff
@@ -459,20 +513,16 @@ function vtkWidgetManager(publicAPI, model) {
...
-
-      if (pickingAvailable) {
-        model.selections = model.selector.generateSelection(x, y, x, y);
+        captureBuffers(x, y, x, y);
       }
     }
   };
 
   publicAPI.updateSelectionFromMouseEvent = (event) => {
+    console.warn(
```
Comment (floryst) [inline Sources/Widgets/Core/WidgetManager/index.js:523]: Ditto with `vtkWarningMacro`

**Review by floryst — COMMENTED** [review]
The changes look good to me overall.

**Pull request merged**